source/core.py

```diff
@@ -274,36 +272,41 @@ class MessageWindow(windowUtils.CustomWindow):
 
 		def __init__(self, windowName=None):
 			super(MessageWindow, self).__init__(windowName)
-			self.handlePowerStatusChange(initializing=True)
+			self.handlePowerStatusChange()
 
 		def windowProc(self, hwnd, msg, wParam, lParam):
-			import ui
-			import winUser
 			if msg == self.WM_POWERBROADCAST and wParam == self.PBT_APMPOWERSTATUSCHANGE:
 				self.handlePowerStatusChange()
 			elif msg == self.WM_DISPLAYCHANGE:
-				# Resolution detection comes from an article found at https://msdn.microsoft.com/en-us/library/ms812142.aspx.
-				#The low word is the width and hiword is height.
-				if winUser.LOWORD(lParam) > winUser.HIWORD(lParam):
-					#Translators: The screen is oriented so that it is wider than it is tall.
-					ui.message(_("Landscape" ))
-				else:
-					#Translators: The screen is oriented in such a way that the height is taller than it is wide.
-					ui.message(_("Portrait"))
-
-		def handlePowerStatusChange(self, initializing=False):
+				self.handleScreenOrientationChange(lParam)
+
+		def handleScreenOrientationChange(self, lParam):
+			import ui
+			import winUser
+			# Resolution detection comes from an article found at https://msdn.microsoft.com/en-us/library/ms812142.aspx.
+			#The low word is the width and hiword is height.
+			if winUser.LOWORD(lParam) > winUser.HIWORD(lParam):
+				#Translators: The screen is oriented so that it is wider than it is tall.
+				ui.message(_("Landscape" ))
+			else:
+				#Translators: The screen is oriented in such a way that the height is taller than it is wide.
+				ui.message(_("Portrait"))
+
+		def handlePowerStatusChange(self):
 			#Mostly taken from script_say_battery_status, but modified.
 			import ui
 			import winKernel
 			sps = winKernel.SYSTEM_POWER_STATUS()
 			if not winKernel.GetSystemPowerStatus(sps) or sps.BatteryFlag is self.UNKNOWN_BATTERY_STATUS:
 				return
-			#This cache is Necessary because sometimes this event double fires, and  if the battery decreases by 3 percent, this event occurs.
-			if sps.BatteryFlag & self.NO_SYSTEM_BATTERY or sps.ACLineStatus == self.oldBatteryStatus:
-				return
-			MessageWindow.oldBatteryStatus = sps.ACLineStatus
-			if  initializing:
+			try:
+				#This cache is Necessary because sometimes this event double fires, and  if the battery decreases by 3 percent, this event occurs.
+				if sps.BatteryFlag & self.NO_SYSTEM_BATTERY or sps.ACLineStatus == self.oldBatteryStatus:
+					return
+			except AttributeError:

```

Comment:
This pattern for checking init is fine, except I'd prefer having the cache initialized to None in the constructor and then testing that here, rather than relying on AttributeError. It just looks cleaner because you know what the initialized state is meant to be. It also avoids hiding a typo if that got changed by mistake in future. :)